Assets/Scripts/Game/Tasks/leaks_riddle/LeaksFactory.cs

```diff
@@ -0,0 +1,67 @@
+using UnityEngine;
+
+namespace Game.Tasks.leaks_riddle
+{
+    public class LeaksFactory : GameTaskFactory<TaskSpawnPoint>
+    {
+        [SerializeField] private GameObject leaksRiddlePrefab;
+        [SerializeField] private Transform[] leakSpawnPoints;
+        [SerializeField] private GameObject leakPrefab;
+        [SerializeField] private Difficulty difficulty;
```

Comment:
```suggestion

```
This reference can be made redundant, if we change the visibility of the attribute `m_Difficulty` in `GameTaskFactory` to `protected` and its name to `difficulty`.
In my opinion, this would be a good idea, as future factories may need the information about the difficulty too.
And the approach is easier to use, as we do not need to add the reference to every factory instance specifically.